src/ansys/sherlock/core/layer.py

```diff
@@ -889,3 +892,74 @@ def export_all_mount_points(
             raise e
 
         return response.value
+
+    def delete_modeling_region(self, project: str, delete_regions: List[Dict[str, str]]):
+        """Delete one or more modeling regions for a specific project.
+
+        Parameters
+        ----------
+        project : str
+            Name of the Sherlock project.
+        delete_regions : list of dict
+            List of modeling regions to delete. Each dictionary should contain:
+            - "cca_name" : str, Name of the CCA.
+            - "region_id" : str, Unique region ID of the modeling region to delete.
+
+        Returns
+        -------
+        int
+            Status code of the response. 0 for success.
+
+        Examples
+        --------
+        >>> from ansys.sherlock.core.launcher import launch_sherlock
+        >>> sherlock = launch_sherlock()
+        >>> sherlock.project.import_odb_archive(
+                "ODB++ Tutorial.tgz",
+                True,
+                True,
+                True,
+                True,
+                project="Test",
+                cca_name="Card",
+            )
+        >>> delete_regions = [{"cca_name": "Card", "region_id": "12345"}]
+        >>> sherlock.layer.delete_modeling_region("Test", delete_regions)
+        """
+        try:
+            if project == "":
+                raise SherlockDeleteModelingRegionError(message="Project name is invalid.")
+            if not delete_regions:
+                raise SherlockDeleteModelingRegionError(message="Delete regions list is empty.")
+
+            for region in delete_regions:

```

Comment:
May want to verify that `region` is a list and add a test for it. This is being done elsewhere.
Otherwise, the error may not be obvious to the end user if they include something that isn't a list.